data/provider/src/main/java/tech/pegasys/teku/api/stateselector/StateSelectorFactory.java

```diff
@@ -80,19 +80,13 @@ public StateSelector genesisSelector() {
   @Override
   public StateSelector finalizedSelector() {
     return () ->
-        SafeFuture.completedFuture(
-            client
-                .getLatestFinalized()
-                .map(
-                    finalized ->
-                        addMetaData(
-                            finalized.getState(),
-                            // The finalized checkpoint may change because of optimistically
-                            // imported blocks at the head and if the head isn't optimistic, the
-                            // finalized block can't be optimistic.
```

Comment:
there was nothing testing this.

i went with the on_block implementation of the finalized state where we base it off of the checkpoint.